kyuubi-server/src/main/scala/org/apache/kyuubi/server/FileCleanupService.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.server
+
+import java.nio.file.Path
+import java.util.concurrent.{Executors, TimeUnit}
+import java.util.concurrent.atomic.AtomicLong
+
+import com.google.common.cache.{Cache, CacheBuilder, RemovalNotification}
+
+import org.apache.kyuubi.Utils
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.service.AbstractService
+import org.apache.kyuubi.util.FileExpirationUtils
+
+class FileCleanupService(name: String) extends AbstractService(name) {
```

Comment:
seems we also need such a service for engine.

Reply:
maybe we can give it a more generic name, like `TempFileManager`

Reply:
OK. Renamed to `TempFileManager` .

I would keep this inside the service module, as
1. it require's Guava cache for eviction and removal listener. the common module does not depend on Guava.
2. server is required to maintain running status for much longer time than engine. so the leaking problem is more urgent in server.